### ISSUE TITLE
Single file mount is not updating when changed on docker host

### ISSUE THREAD

**Issue by amireldor** [docker/for-win#328]
(note: mounted folders work just fine, files getting updated and all)

### Expected behavior

Mount a single file on Windows to a container, update the file on Windows, see changes on the file inside the container.

### Actual behavior

I mount a single file from Windows to a Linux container. When I update the file on Windows, the changes are **not** reflected in the Linux container. This is not related to some funky text editor (Vim, etc...), with notepad it wouldn't update as well.

### Steps to reproduce

This works bad:

    docker run --rm -it -v ${PWD}/temp.txt:/temp.txt alpine /bin/sh
    # changes to temp.txt (Windows) do not appear in /temp.txt (Linux)

On the other hand, this works:

    docker run --rm -it -v ${PWD}:/docker-is-fun alpine /bin/sh
    # this works. Changes to temp.txt (Windows) are reflected in /docker-is-fun/temp.txt

**Comment by dgageot** [comment]
@amireldor Which version of Docker for Windows are you using? Can you share a diagnostic id so that we can investigate?

**Comment by amireldor** [comment]
I'm using the stable version and this is the diagnostic id: 22270F62-32B5-4086-A6BB-8C6AB5E4F858/2016-12-18_22-52-12

Stranger things happened, but I think I'm starting to get a clue...

On a different admin user on the same machine, I ran similar commands as I mentioned above. It seemed as the volumes I mounted on the ephemeral containers kept a persistent state which is different than the actual folder I have on the host. So I realized I am using Docker volumes instead of mounting an existing folder from the Docker host.

After that I used `docker volume ls` and `docker volume rm` to delete the 3 accidental volumes created by my previous experiments, I fired up another container with a commands I used before (up-arrow a few times on the command prompt) I saw that I still have the same mounted folders and their contents (it's worth mentioning I created several files from within the container). I assume they are in the same accidental volume as before which was not removed properly :( (no volumes listed with `docker volume ls`).
